dsgrid/registry/project_registry_manager.py

```diff
@@ -300,6 +441,19 @@ def _register(self, config, submitter, log_message, force):
 
     @track_timing(timer_stats_collector)
     def _run_checks(self, config: ProjectConfig):
+        dims = []
+        dims_by_type = defaultdict(list)
+        for dim in config.iter_dimensions():
+            dims.append(dim)
+            dims_by_type[dim.model.dimension_type].append(dim)
+
+        check_uniqueness((x.model.name for x in dims), "dimension name")
+        for dims in dims_by_type.values():
+            check_uniqueness((x.model.display_name for x in dims), "dimension display name")
+        check_uniqueness(
+            (getattr(x.model, "cls") for x in config.model.dimensions.base_dimensions),
+            "dimension cls",
+        )
```

Comment:
what about check_uniqueness of query_names? I know most query_names are just a modified version of model_names, but we are allowing user-input query_names in the config.